sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// Represents a policy that can be overriden to customize whether or not a request will be retried and how long to wait before retrying.
+    /// </summary>
+    public abstract class RetryPolicy : HttpPipelinePolicy
+    {
+        private readonly RetryMode _mode;
+        private readonly TimeSpan _delay;
+        private readonly TimeSpan _maxDelay;
+        private readonly int _maxRetries;
+
+        private readonly Random _random = new ThreadSafeRandom();
```

Comment:
I don't think we would want that as it could cause more contention between different client/pipeline instances. I do think this type can be improved though to avoid using locks similar to what we do in SB/EH - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs#L53